container-chains/templates/frontier/runtime/src/lib.rs

```diff
@@ -1208,23 +1217,82 @@ impl_runtime_apis! {
                 }
 
                 fn reserve_transferable_asset_and_dest() -> Option<(MultiAsset, MultiLocation)> {
+                    use xcm_config::SelfReserve;
                     // AH can reserve transfer native token to some random parachain.
                     let random_para_id = 43211234;
+                    let balance = EXISTENTIAL_DEPOSIT * 10;
+
                     ParachainSystem::open_outbound_hrmp_channel_for_benchmarks_or_tests(
                         random_para_id.into()
                     );
                     Some((
                         MultiAsset {
-                            fun: Fungible(EXISTENTIAL_DEPOSIT),
-                            id: Concrete(Parent.into())
+                            fun: Fungible(balance),
+                            id: Concrete(SelfReserve::get())
                         },
                         ParentThen(Parachain(random_para_id).into()).into(),
                     ))
                 }
 
                 fn set_up_complex_asset_transfer(
                 ) -> Option<(MultiAssets, u32, MultiLocation, Box<dyn FnOnce()>)> {
-                    None
+                    use xcm_config::SelfReserve;
+                    use sp_runtime::traits::StaticLookup;
+                    // Transfer to Relay some local AH asset (local-reserve-transfer) while paying
+                    // fees using teleported native token.
+                    // (We don't care that Relay doesn't accept incoming unknown AH local asset)
+                    let dest = Parent.into();
+
+                    let fee_amount = EXISTENTIAL_DEPOSIT;
+                    let fee_asset: MultiAsset = (SelfReserve::get(), fee_amount).into();
+
+                    let who = frame_benchmarking::whitelisted_caller();
+                    // Give some multiple of the existential deposit
+                    let balance = fee_amount + EXISTENTIAL_DEPOSIT * 1000;
+                    log::info!("setting {:?}", EXISTENTIAL_DEPOSIT);
+                    let _ = <Balances as frame_support::traits::Currency<_>>::make_free_balance_be(
+                        &who, balance,
+                    );
+
+                    // verify initial balance
+                    assert_eq!(Balances::free_balance(&who), balance);
+
+                    // set up local asset
+                    let asset_amount = 10u128;
+                    let initial_asset_amount = asset_amount * 10;
+                    let asset_id = 0;
+
+                    // inject it into pallet-foreign-asset-creator.
+                    let asset_location = MultiLocation::new(
+                        0,
+                        X2(PalletInstance(50), GeneralIndex(u32::from(asset_id).into()))
```

Comment:
Can we define some constant and use it? That would ensure nobody accidentally modify this.

Reply:
I will look into this

Reply:
would you want this to be somewhere common to all runtimes? If so, I think this would leave in dancekit, and I would need to cherry-pick the change here. Maybe it's a cleanup we can do in a follow up?

Reply:
My idea actually is that we generate in foreign asset creator a benchmarking code that is called: `register_foreign_asset_for_benchmarking` which would abstract all this. But again I think we can leave this for future PR